pandas/core/series.py

```diff
@@ -5037,12 +5041,42 @@ def between(self, left, right, inclusive=True) -> Series:
         3    False
         dtype: bool
         """
-        if inclusive:
+
+        if inclusive == "both":
             lmask = self >= left
             rmask = self <= right
-        else:
+        elif inclusive == "left":
+            lmask = self >= left
+            rmask = self < right
+        elif inclusive == "right":
+            lmask = self > left
+            rmask = self <= right
+        elif inclusive == "neither":
             lmask = self > left
             rmask = self < right
+        elif inclusive is False:
+            warnings.warn(
+                "Boolean inputs to the `inclusive` argument are deprecated in"
+                "favour of `both` or `neither`.",
+                FutureWarning,
+                stacklevel=2,
+            )
+            lmask = self > left
+            rmask = self < right
```

Comment:
I think maybe better to have `if inclusive == "both" or inclusive is True`  (and maybe keep the default as True) so that we ensure that the this works as before until the deprecation is actually enforced. (and the same for 'inclusive=False`)

Reply:
I see @jreback already commented on this https://github.com/pandas-dev/pandas/pull/40628#discussion_r629419084

can you move the True/False cases to be first and translate the values.

Reply:
Alright I see what you guys mean now; I misunderstood the previous comment. I changed it so that the True/False cases are handled and translated first.